src/software/thunderscope/gl/layers/gl_movement_field_test_layer.py

```diff
@@ -0,0 +1,82 @@
+import math
+from pyqtgraph.Qt.QtCore import Qt
+from software.thunderscope.gl.helpers.extended_gl_view_widget import MouseInSceneEvent
+from proto.import_all_protos import *
+from software.thunderscope.proto_unix_io import ProtoUnixIO
+from software.logger.logger import create_logger
+from software.thunderscope.gl.layers.gl_layer import GLLayer
+from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
+
+logger = create_logger(__name__)
+
+
+class GLMovementFieldTestLayer(GLLayer):
+    def __init__(
+        self, name: str, fullsystem_io: ProtoUnixIO, buffer_size: int = 5
+    ) -> None:
+        """Initialize the GLMovementLayer
+
+        :param name:            The displayed name of the layer
+        :param buffer_size:     The buffer size we have
+        :param fullsystem_io:   The fullsystem protounix io
+        """
+        super().__init__(name)
+
+        self.world_buffer: ThreadSafeBuffer = ThreadSafeBuffer(buffer_size, World)
+        self.fullsystem_io: ProtoUnixIO = fullsystem_io
+        self.friendly_robot_on_field = []
+
+    def mouse_in_scene_pressed(self, event: MouseInSceneEvent) -> None:
+        """Move to the point clicked
+
+        :param event: The event
+        """
+        if not self.visible():
+            return
+
+        if not event.mouse_event.modifiers() == Qt.KeyboardModifier.ShiftModifier:
+            return
+
+        point = event.point_in_scene
+        logger.info(f"I am clicking on point: {point}")
+        self.move_to_point(point)
+
+    def move_to_point(self, point):
+        """Move to a point
+
+        :param point: the point we are commanding the robot to move to
+        """
+        if len(self.friendly_robot_on_field) == 0:
+            logger.warning("There are no friendly robots on the field.")
+            return
+
+        robot_id = self.friendly_robot_on_field[0]
```

Comment:
maybe it should use the closest robot to the point?

Reply:
would we want to move specific robots to various points? maybe we could have a field / dropdown where we select the robot to move first, and then every shift-click after that moves that robot only

Reply:
Or you could implement as shift click to select, then click to move?